speaking-portal-project/src/test/file_check.ts

```diff
@@ -64,45 +68,65 @@ export function checkFileReadAccess(file: string) {
      * @param file - The name of the file to check
      */
     fs.access(file, fs.constants.R_OK, (err) => {
-        // TODO: should we separate these checks?
         if (err) throw Error(`FileNotReadable`)
-        else console.log(`${file} is readable`)
-        return true
     })
+    console.log(`${file} is readable`)
+    return true
 }
 
-export function checkWavFile(file: string) {
-    // TODO: Should we be checking the name that is given or look for a .wav file in the directory?
+export async function isWavFile(file: string) {
+    /**
+     * Checks if the specified audio file is .wav
+     *
+     * @remarks
+     * Ensure checkFileExistence() and checkFileReadAccess() are completed prior to this check
+     *
+     * @param file - The name of the file to check
+     */
     if (file.toLowerCase().includes('.wav')) {
-        console.log(`Audio file ${file} has .wav extension.`)
+        console.log(`Audio file ${file} has .wav extension`)
     } else {
-        throw Error(`Supposed audio file ${file} is not a .wav file.`)
+        throw Error(`AudioNotWavFile`)
     }
+    return true
+}
 
-    // TODO: Check if files are valid/not corrupted
-    // should we do this manually or with this module? https://www.npmjs.com/package/wav-file-info
-
-    // TODO: check if audio is 0 seconds long
+export async function isWavFileValid(file: string) {
+    /**
+     * Checks if the specified .wav file is corrupted
+     *
+     * @remarks
+     * Ensure checkFileExistence(), checkFileReadAccess(), and checkWavFile() are completed prior to this check
+     *
+     * @param file - The name of the file to check
+     */
+    wavFileInfo.infoByFilename(file, function(err:any, info:any){
```

Comment:
Future we could somehow type out the error 'type', rather than using 'any' here. Strange this module as not imported types for this function